crawl-ref/source/makeitem.cc

```diff
@@ -259,24 +258,15 @@ static bool _try_make_weapon_artefact(item_def& item, int force_type,
         item.plus -= biased_random2(7,2);
         item.plus -= biased_random2(7,2);
 
-        bool cursed = false;
         if (one_chance_in(5))
-        {
-            cursed = true;
-            item.plus = 3 - random2(6);
-        }
-        else if (item.plus < 0 && !one_chance_in(3))
-            cursed = true;
+            item.plus = 3 - random2(6); // not the best
```

Comment:
Well, this can't produce negative enchantments. See directly below this:

```
        if (one_chance_in(5))
            item.plus = 3 - random2(6); // not the best

        // On weapons, an enchantment of less than 0 is never viable.
        item.plus = max(static_cast<int>(item.plus), random2(2));
```

I'm fine removing this logic, if you still think it's best (and I think you could be right), but just want to clarify here.

Reply:
Actually, you know what, I'll just remove this.